model/robot_test_results.go

```diff
@@ -0,0 +1,79 @@
+package model
+
+//go:generate generator
+
+type RobotTestResults struct {
+	Failed        bool
+	ReachedFinish bool
+	Moves         TypedPositions
+}
+
+const (
+	Up = iota
+	Right
+	Down
+	Left
+)
+
+type Position struct {
+	X,
+	Y int
+}
+
+func (p Position) Equals(pos Position) bool {
+	return p.X == pos.X && p.Y == pos.Y
+}
+
+type TypedPosition struct {
+	Position
+	Type string
```

Comment:
You have some magic strings in your code. Please make `Type` a type (`type Type string`) and define constants, just like you did for the directions (`Up`, `Right`, ...).